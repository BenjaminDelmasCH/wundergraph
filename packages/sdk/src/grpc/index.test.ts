import { protosetToGrpcApiObject } from './index';
import * as fs from 'fs';

const runTest = async (testFile: string, snapShot: string) => {
	const exists = fs.existsSync(testFile);
	if (!exists) {
		return;
	}
	const fileContents = fs.readFileSync(testFile).toString('base64');

	const actual = await protosetToGrpcApiObject(fileContents);

	expect(actual.Schema).toMatchSnapshot(snapShot + '_' + 'schema');
	expect(JSON.stringify(actual.DataSources, null, 2)).toMatchSnapshot(snapShot + '_' + 'data_sources');
	expect(JSON.stringify(actual.Fields, null, 2)).toMatchSnapshot(snapShot + '_' + 'fields');

	// const content = JSON.stringify(actual, null, ' ')
	// fs.writeFileSync('src/grpc/'+snapShot + '.json', content);
};

test('starwars grpc', async () => {
	await runTest('../../pkg/datasources/grpc/testdata/starwars/starwars.protoset', 'starwars_grpc');
});
