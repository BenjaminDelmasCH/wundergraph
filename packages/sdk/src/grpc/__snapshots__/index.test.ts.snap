// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`starwars grpc: starwars_grpc_data_sources 1`] = `
"[
  {
    \\"Kind\\": 8,
    \\"RootNodes\\": [
      {
        \\"typeName\\": \\"Query\\",
        \\"fieldNames\\": [
          \\"grpc_starwars_StarwarsService_GetHero\\"
        ]
      }
    ],
    \\"ChildNodes\\": [],
    \\"Custom\\": {
      \\"server\\": {
        \\"protoset\\": \\"CroICg5zdGFyd2Fycy5wcm90bxIIc3RhcndhcnMigAIKCUNoYXJhY3RlchIOCgJpZBgBIAEoCVICaWQSEgoEbmFtZRgCIAEoCVIEbmFtZRItCgdmcmllbmRzGAMgAygLMhMuc3RhcndhcnMuQ2hhcmFjdGVyUgdmcmllbmRzEjAKCmFwcGVhcnNfaW4YBCADKA4yES5zdGFyd2Fycy5FcGlzb2RlUglhcHBlYXJzSW4SHwoLaG9tZV9wbGFuZXQYBSABKAlSCmhvbWVQbGFuZXQSKQoQcHJpbWFyeV9mdW5jdGlvbhgGIAEoCVIPcHJpbWFyeUZ1bmN0aW9uEiIKBHR5cGUYByABKA4yDi5zdGFyd2Fycy5UeXBlUgR0eXBlIj0KDkdldEhlcm9SZXF1ZXN0EisKB2VwaXNvZGUYASABKA4yES5zdGFyd2Fycy5FcGlzb2RlUgdlcGlzb2RlIiEKD0dldEh1bWFuUmVxdWVzdBIOCgJpZBgBIAEoCVICaWQiIQoPR2V0RHJvaWRSZXF1ZXN0Eg4KAmlkGAEgASgJUgJpZCISChBMaXN0RW1wdHlSZXF1ZXN0IkEKEkxpc3RIdW1hbnNSZXNwb25zZRIrCgZodW1hbnMYASADKAsyEy5zdGFyd2Fycy5DaGFyYWN0ZXJSBmh1bWFucyJBChJMaXN0RHJvaWRzUmVzcG9uc2USKwoGZHJvaWRzGAEgAygLMhMuc3RhcndhcnMuQ2hhcmFjdGVyUgZkcm9pZHMqHAoEVHlwZRIJCgVIVU1BThAAEgkKBURST0lEEAEqMwoHRXBpc29kZRIFCgFfEAASCwoHTkVXSE9QRRABEgoKBkVNUElSRRACEggKBEpFREkQAzLTAgoPU3RhcndhcnNTZXJ2aWNlEjgKB0dldEhlcm8SGC5zdGFyd2Fycy5HZXRIZXJvUmVxdWVzdBoTLnN0YXJ3YXJzLkNoYXJhY3RlchI6CghHZXRIdW1hbhIZLnN0YXJ3YXJzLkdldEh1bWFuUmVxdWVzdBoTLnN0YXJ3YXJzLkNoYXJhY3RlchI6CghHZXREcm9pZBIZLnN0YXJ3YXJzLkdldERyb2lkUmVxdWVzdBoTLnN0YXJ3YXJzLkNoYXJhY3RlchJGCgpMaXN0SHVtYW5zEhouc3RhcndhcnMuTGlzdEVtcHR5UmVxdWVzdBocLnN0YXJ3YXJzLkxpc3RIdW1hbnNSZXNwb25zZRJGCgpMaXN0RHJvaWRzEhouc3RhcndhcnMuTGlzdEVtcHR5UmVxdWVzdBocLnN0YXJ3YXJzLkxpc3REcm9pZHNSZXNwb25zZUJLWklnaXRodWIuY29tL3d1bmRlcmdyYXBoL3d1bmRlcmdyYXBoL3BrZy9kYXRhc291cmNlcy9ncnBjL3Rlc3RkYXRhL3N0YXJ3YXJzYgZwcm90bzM=\\",
        \\"target\\": {
          \\"kind\\": 0,
          \\"staticVariableContent\\": \\"127.0.0.1:9095\\",
          \\"environmentVariableName\\": \\"\\",
          \\"environmentVariableDefaultValue\\": \\"\\",
          \\"placeholderVariableName\\": \\"\\"
        }
      },
      \\"endpoint\\": {
        \\"package\\": \\"starwars\\",
        \\"service\\": \\"StarwarsService\\",
        \\"method\\": \\"GetHero\\"
      },
      \\"request\\": {
        \\"header\\": {},
        \\"body\\": \\"{{ .arguments.input }}\\"
      }
    },
    \\"Directives\\": []
  }
]"
`;

exports[`starwars grpc: starwars_grpc_fields 1`] = `
"[
  {
    \\"typeName\\": \\"Query\\",
    \\"fieldName\\": \\"grpc_starwars_StarwarsService_GetHero\\",
    \\"disableDefaultFieldMapping\\": true,
    \\"path\\": [],
    \\"argumentsConfiguration\\": [],
    \\"requiresFields\\": [],
    \\"unescapeResponseJson\\": false
  }
]"
`;

exports[`starwars grpc: starwars_grpc_schema 1`] = `
"type Query {
  grpc_starwars_StarwarsService_GetHero(input: grpc_starwars_GetHeroRequest_Input): grpc_starwars_Character
  grpc_starwars_StarwarsService_GetHuman(input: grpc_starwars_GetHumanRequest_Input): grpc_starwars_Character
  grpc_starwars_StarwarsService_GetDroid(input: grpc_starwars_GetDroidRequest_Input): grpc_starwars_Character
  grpc_starwars_StarwarsService_ListHumans(input: grpc_starwars_ListEmptyRequest_Input): grpc_starwars_ListHumansResponse
  grpc_starwars_StarwarsService_ListDroids(input: grpc_starwars_ListEmptyRequest_Input): grpc_starwars_ListDroidsResponse
  grpc_starwars_StarwarsService_connectivityState(tryToConnect: Boolean): grpc_ConnectivityState
}

type grpc_starwars_Character {
  id: String
  name: String
  friends: [grpc_starwars_Character]
  appears_in: [grpc_starwars_Episode]
  home_planet: String
  primary_function: String
  type: grpc_starwars_Type
}

enum grpc_starwars_Episode {
  _
  NEWHOPE
  EMPIRE
  JEDI
}

enum grpc_starwars_Type {
  HUMAN
  DROID
}

input grpc_starwars_GetHeroRequest_Input {
  episode: grpc_starwars_Episode
}

input grpc_starwars_GetHumanRequest_Input {
  id: String
}

input grpc_starwars_GetDroidRequest_Input {
  id: String
}

type grpc_starwars_ListHumansResponse {
  humans: [grpc_starwars_Character]
}

scalar grpc_starwars_ListEmptyRequest_Input @specifiedBy(url: \\"http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf\\")

type grpc_starwars_ListDroidsResponse {
  droids: [grpc_starwars_Character]
}

enum grpc_ConnectivityState {
  IDLE
  CONNECTING
  READY
  TRANSIENT_FAILURE
  SHUTDOWN
}"
`;
